dask/array/tests/test_routines.py

```diff
@@ -1251,6 +1251,42 @@ def test_nonzero_method():
             assert_eq(d_nz[i], x_nz[i])
 
 
+@pytest.mark.skipif(
+    LooseVersion(np.__version__) < LooseVersion("1.14.0"),
+    reason="NumPy 1.14.0+ needed for `unravel_index` to take an empty shape."
+)
+def test_unravel_index_empty():
+    shape = tuple()
+    findices = np.array(0, dtype=int)
+    d_findices = da.from_array(findices, chunks=1)
+
+    indices = np.unravel_index(findices, shape)
+    d_indices = da.unravel_index(d_findices, shape)
+
+    assert isinstance(d_indices, type(indices))
+    assert len(d_indices) == len(indices) == 0
+
+
+def test_unravel_index():
+    for nindices, shape, order in [(0, (15,), 'C'),
```

Comment:
FYI pytest.parametrize helps you quickly find which case is failing.
https://docs.pytest.org/en/latest/parametrize.html

It saves me a bunch of headaches.

Reply:
Yeah we've gone back and forth between using `parametrize`. Pro: it makes it really easy to write a bunch of tests, be sure they get covered, and see exactly which combinations are failing. Con: it adds a lot of noise when running the tests (particularly in verbose mode). Given they are generally passing after figuring out how to get things to work, this noise complaint becomes a bit more significant.

In this case, I think I stole this configuration from above. Personally I prefer `parametrize` and would be happy to switch though.

Reply:
no need to switch. Just like future note you know. it is equivalent. who really cares until a bug shows up.